src/menu_events.cpp

```diff
@@ -701,14 +701,14 @@ void create_and_place(game_display&,
 		unit_map&,
 		const map_location& loc,
 		const unit_type& u_type,
-		unit_race::GENDER gender = unit_race::NUM_GENDERS)
+		const unit_gender* gender = nullptr)
 {
 	synced_context::run_and_throw("debug_create_unit",
 		config {
 			"x", loc.wml_x(),
 			"y", loc.wml_y(),
 			"type", u_type.id(),
-			"gender", gender_string(gender),
+			"gender", gender ? gender->str() : unit_gender::male().str(),
```

Comment:
This sort of thing also bugs me a little. Maybe it's not exactly best practice, but couldn't we do something like this?

```

std::string unit_gender::str() {
    if(this == nullptr) return unit_gender::male().str();
    // normal implementation of str() here
}
```

Reply:
we can't. Calling a memeber function on a nullptr is undefioned behviour, in particular the compiler might simply assume 'this != nullptr' and optimize that check out.